infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -78,30 +78,35 @@ public FailureDetector(String localEndpoint) {
      * @param layout Current Layout
      */
     public PollReport poll(
-            @Nonnull Layout layout, @Nonnull CorfuRuntime corfuRuntime, @NonNull SequencerMetrics sequencerMetrics) {
+            @Nonnull Layout layout, @Nonnull CorfuRuntime corfuRuntime, @NonNull SequencerMetrics sequencerMetrics,
+            FileSystemStats fileSystemStats) {
 
         log.trace("Poll report. Layout: {}", layout);
 
-        Map<String, IClientRouter> routerMap;
-
         // Collect and set all responsive servers in the members array.
         Set<String> allServers = layout.getAllServers();
 
-        // Set up arrays for routers to the endpoints.
-        routerMap = new HashMap<>();
-        allServers.forEach(s -> {
-            IClientRouter router = corfuRuntime.getRouter(s);
-            router.setTimeoutResponse(networkStretcher.getCurrentPeriod().toMillis());
-            routerMap.put(s, router);
-        });
+        Map<String, IClientRouter> routers = adjustRouters(corfuRuntime, allServers);
 
         // Perform polling of all responsive servers.
         return pollRound(
-                layout.getEpoch(), layout.getClusterId(), allServers, routerMap, sequencerMetrics,
-                ImmutableList.copyOf(layout.getUnresponsiveServers())
+                layout.getEpoch(), layout.getClusterId(), allServers, routers, sequencerMetrics,
+                ImmutableList.copyOf(layout.getUnresponsiveServers()),
+                fileSystemStats
         );
     }
 
+    private Map<String, IClientRouter> adjustRouters(CorfuRuntime corfuRuntime, Set<String> allServers) {
```

Comment:
I think we are at a stage where there is a common understanding - or perhaps not - that adjusting a timeout per-ping is completely unnecessary and only complicates the logic and reasoning about correctness, I suggest we just nuke it.

Reply:
Maybe, but that's outside of this activity of bringing file system failures into the failure detector. I'd be happy to change that in other pr-s. 
P.S. `adjustRouters` staying there since the beginning, I just moved that piece of code into a separate method.